src/app/containers/ATIAnalytics/atiUrl/index.js

```diff
@@ -42,6 +42,15 @@ export const buildATIPageTrackPath = ({
   categoryName,
   campaigns,
 }) => {
+  const href = getHref(platform);
+  const referrer = getReferrer(platform, origin, previousPath);
+
+  // We use amp variable substitutes to get the href and referrer and these cannot be manipulated
```

Comment:
OK cool - so `getHref` and `getReferrer` return the AMP substitution variable on AMP, and use JS in the browser for canonical. We don't want to make any updates in those functions because they are used elsewhere without this 'double encoding' requirement.

Reply:
>  We don't want to make any updates in those functions because they are used elsewhere without this 'double encoding' requirement.

only `getReferrer` is used elsewhere (used for chartbeat) but I thought since the requirement is quite specific to x6 and x5 that I would put it here.